File: 2D-World-Simulator-Java/OPP-Java/src/op/JavaProject/Organisms/Plants/Belladonna.java
package op.JavaProject.Organisms.Plants;

import op.JavaProject.Functionality.BoardPoint;
import op.JavaProject.Functionality.Species;
import op.JavaProject.Game.Organism;
import op.JavaProject.Game.Plant;
import op.JavaProject.Game.World;

import java.awt.*;

public class Belladonna extends Plant {

    public Belladonna(World organismWorld, BoardPoint point) {
        super(organismWorld, point);
        this.strength = 99;
        this.initiative = 0;
        this.organismSign = "O";
        this.organismSpecie = Species.GRASS;
        this.organismColor = new Color(156, 127, 230);
    }

    //checking defence successfulness
    @Override
    public boolean Defence(Organism attackingOrganism) {

        if (KillAnimal(attackingOrganism)) return true;
        return false;
    }

    //performing defence -> killing animal
    private boolean KillAnimal(Organism attackingOrganism) {
        organismWorld.DeleteOrganism(attackingOrganism);
        organismWorld.GetCommentator().AddComment(attackingOrganism.GetOrganismSign() + " died by eating BELLADONNA!");
        return true;
    }

}

File: 2D-World-Simulator-Python/OPP-Python/Organisms/Plants/Belladonna.py
from Game.OrganismStatics import *
from Game.Plant import Plant


class Belladonna(Plant):
    def __init__(self, organism_world, point):
        super().__init__(organism_world, point, 99, 0, "0", OrganismSpecie.BELLADONNA, "#9c7fe6")

    # checking defence successfulness
    def defence(self, attacking_organism):
        # always performing defence -> killing animal
        self.organism_world.delete_organism(attacking_organism)
        self.organism_world.get_world_window().add_comment(
            attacking_organism.get_organism_sign() + " died by eating BELLADONNA!")
        return True
